pkg/util/yaml/decoder.go

```diff
@@ -217,9 +233,13 @@ func (d *YAMLOrJSONDecoder) Decode(into interface{}) error {
 			js := string(data)
 			start := strings.LastIndex(js[:syntax.Offset], "\n") + 1
 			line := strings.Count(js[:start], "\n")
-			return fmt.Errorf("json: line %d: %s", line, syntax.Error())
+			return JSONSyntaxError{fmt.Errorf("json: line %d: %s", line, syntax.Error())}
 		}
 	}
+
+	if err != nil && err != io.EOF && hasPrefix([]byte(err.Error()), []byte("yaml: line")) {
+		return YAMLSyntaxError{err}
+	}
```

Comment:
@smarterclayton unfortunately the `yaml` package does not contain a `SyntaxError` type, so I am handling yaml syntax errors based on their prefix. PTAL